stubs/D3DShot/d3dshot/dll/d3d.pyi

```diff
@@ -0,0 +1,219 @@
+import sys
+from _typeshed import Incomplete
+from collections.abc import Callable
+from ctypes import Structure, _CArgObject, c_int32, c_uint, c_void_p, wintypes
+from typing_extensions import TypeAlias
+
+from d3dshot.dll.dxgi import IDXGIAdapter
+
+# mypy does not support os.name checks, while pyright does https://github.com/python/mypy/issues/13002
+# import os
+# if os.name == "nt":  # noqa: Y002
+if sys.platform == "win32":
+    from ctypes import HRESULT
+
+    _HRESULT: TypeAlias = HRESULT
+else:
+    _HRESULT: TypeAlias = Incomplete
+
+# TODO: Complete types once we can import non-types dependencies
+# See: https://github.com/python/typeshed/issues/5768
+# import comtypes
+_IUnknown: TypeAlias = Incomplete
+
+class DXGI_SAMPLE_DESC(Structure):
+    Count: wintypes.UINT
+    Quality: wintypes.UINT
+
+class D3D11_BOX(Structure):
+    left: wintypes.UINT
+    top: wintypes.UINT
+    front: wintypes.UINT
+    right: wintypes.UINT
+    bottom: wintypes.UINT
+    back: wintypes.UINT
+
+class D3D11_TEXTURE2D_DESC(Structure):
+    Width: wintypes.UINT
+    Height: wintypes.UINT
+    MipLevels: wintypes.UINT
+    ArraySize: wintypes.UINT
+    Format: wintypes.UINT
+    SampleDesc: DXGI_SAMPLE_DESC
+    Usage: wintypes.UINT
+    BindFlags: wintypes.UINT
+    CPUAccessFlags: wintypes.UINT
+    MiscFlags: wintypes.UINT
+
+class ID3D11DeviceChild(_IUnknown):
+    GetDevice: Callable[[], None]
```

Comment:
Callable attributes tend to cause problems on mypy; maybe check that these behave as you expect.

Reply:
This is probably better as a `def`, that's just 3AM melted brain not recognizing that.